File: 4107-find-missing-elements/find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
          sort(nums.begin(), nums.end());

        set<int> st;
        for (int x : nums) {
            st.insert(x);
        }

        int smallest = nums.front();
        int largest = nums.back();

        vector<int> ans;

        for (int i = smallest; i <= largest; i++) {
            if (st.find(i) == st.end()) {
                ans.push_back(i);
            }
        }

        return ans;
    }
};

File: 4107-find-missing-elements/find-missing-elements.py
# class Solution(object):
#     def findMissingElements(self, nums):
#        n=len(nums)
#        nums.sort()
#        st=set()
#        for i in range(len(nums)):
#             st.add(nums[i])

#        smallest=nums[0]
#        largest=nums[-1]
#        ans = []
#        for i in range (smallest,largest+1):
#           if i  not in st:
#             ans.append(i)

#         return ans
class Solution(object):
    def findMissingElements(self, nums):
        n = len(nums)
        nums.sort()
        st = set()
        for i in range(len(nums)):
            st.add(nums[i])

        smallest = nums[0]
        largest = nums[-1]
        ans = []
        for i in range(smallest, largest + 1):
            if i not in st:
                ans.append(i)

        return ans

            
       
        
        